backend/common/services/simulation_queue_service.py

```
@@ -0,0 +1,383 @@
+    job_data_str = redis_manager.get(job_data_key)
+    
+    if not job_data_str:
+        logger.warning(
+            f"[SIMULATION_QUEUE] Job data not found for job_id={job_id}, "
+            f"re-queuing job to prevent permanent loss"
+        )
+        # Re-queue the job ID to prevent permanent loss
+        try:
+            redis_manager.lpush(QUEUE_KEY, job_id)
+            logger.info(f"[SIMULATION_QUEUE] Re-queued job_id={job_id} after missing data")
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)